ledger-core/v2/cmd/pulsewatcher/emoji.go

```diff
@@ -21,8 +21,13 @@ type Emoji struct {
 func NewEmoji() *Emoji {
 	return &Emoji{
 		registred: make(map[uint32]string),
-		light:     []string{"😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️"},
-		virtual:   []string{"⚽", "🏀", "🏈", "🏐", "🏉", "🚗", "🚕", "🚙", "🚌", "🚒", "🚛", "🚜", "🚑️", "🚎", "🏎", "🚐", "🚚"},
+		light:     []string{"😀"},
+		virtual: []string{
+			"😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️", "⚽", "🏀", "🏈", "🏐", "🏉", "🚗", "🚕", "🚙", "🚌", "🚒", "🚛", "🚜", "🚑️", "🚎", "🏎", "🚐", "🚚", "😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️", "😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️", "😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️",
+			"😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️", "⚽", "🏀", "🏈", "🏐", "🏉", "🚗", "🚕", "🚙", "🚌", "🚒", "🚛", "🚜", "🚑️", "🚎", "🏎", "🚐", "🚚", "😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️", "😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️", "😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️",
+			"😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️", "⚽", "🏀", "🏈", "🏐", "🏉", "🚗", "🚕", "🚙", "🚌", "🚒", "🚛", "🚜", "🚑️", "🚎", "🏎", "🚐", "🚚", "😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️", "😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️", "😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️",
+			"😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️", "⚽", "🏀", "🏈", "🏐", "🏉", "🚗", "🚕", "🚙", "🚌", "🚒", "🚛", "🚜", "🚑️", "🚎", "🏎", "🚐", "🚚", "😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️", "😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️", "😀", "😆", "😎", "😭", "😴", "♈️", "♉️", "♊️️", "♋️", "♌️", "♍️", "♎️", "♏️", "♐️", "♑️", "️♒️", "♓️",
+		},
```

Comment:
This is kinda ugly. What'll happen when we go over this limit? I suggest writing an algorithm to iterate multiple times over these emoji arrays. Or at least add a TODO comment there.

Reply:
Pulsewatcher becomes too inconvenient on nodes count >20, maybe we need to think about replacing it, rather then patching

Reply:
done